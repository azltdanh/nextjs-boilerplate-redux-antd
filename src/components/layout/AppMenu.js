import { useRouter } from 'next/router';
import { Menu } from 'antd';

const AppMenu = () => {
  const router = useRouter();

  return (
    <Menu
      mode="horizontal"
      defaultSelectedKeys={['/']}
      selectedKeys={[router.pathname]}
    >
      <Menu.Item
        key="/how-it-works"
        onClick={() => router.push('/how-it-works')}
      >
        HOW IT WORKS
      </Menu.Item>
      <Menu.Item key="/about-us" onClick={() => router.push('/about-us')}>
        ABOUT US
      </Menu.Item>
      <Menu.Item key="/blogs" onClick={() => router.push('/blogs')}>
        BLOGS
      </Menu.Item>
      <Menu.Item key="/contact-us" onClick={() => router.push('/contact-us')}>
        CONTACT US
      </Menu.Item>
      <Menu.Item key="/listing" onClick={() => router.push('/listing')}>
        LISTING
      </Menu.Item>
    </Menu>
  );
};

export default AppMenu;
