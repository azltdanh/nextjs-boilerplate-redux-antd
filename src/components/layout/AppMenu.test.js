import AppMenu from './AppMenu';

const mockHistoryPush = jest.fn();

jest.mock('next/router', () => ({
  ...jest.requireActual('next/router'),
  useRouter: () => ({
    pathname: '/',
    push: mockHistoryPush,
  }),
}));

describe(AppMenu.name, () => {
  it('renders', () => {
    const wrapper = shallow(<AppMenu />);
    expect(wrapper).toMatchSnapshot();

    const items = wrapper.find('MenuItem');
    expect(items.length).toEqual(5);

    items.at(0).simulate('click');
    expect(mockHistoryPush).toHaveBeenCalledWith('/how-it-works');

    items.at(1).simulate('click');
    expect(mockHistoryPush).toHaveBeenCalledWith('/about-us');

    items.at(2).simulate('click');
    expect(mockHistoryPush).toHaveBeenCalledWith('/blogs');

    items.at(3).simulate('click');
    expect(mockHistoryPush).toHaveBeenCalledWith('/contact-us');

    items.at(4).simulate('click');
    expect(mockHistoryPush).toHaveBeenCalledWith('/listing');
  });
});
