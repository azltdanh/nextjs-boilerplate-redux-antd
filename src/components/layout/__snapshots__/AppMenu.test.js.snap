// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppMenu renders 1`] = `
<Menu
  defaultSelectedKeys={
    Array [
      "/",
    ]
  }
  mode="horizontal"
  selectedKeys={
    Array [
      "/",
    ]
  }
>
  <MenuItem
    key="/how-it-works"
    onClick={[Function]}
  >
    HOW IT WORKS
  </MenuItem>
  <MenuItem
    key="/about-us"
    onClick={[Function]}
  >
    ABOUT US
  </MenuItem>
  <MenuItem
    key="/blogs"
    onClick={[Function]}
  >
    BLOGS
  </MenuItem>
  <MenuItem
    key="/contact-us"
    onClick={[Function]}
  >
    CONTACT US
  </MenuItem>
  <MenuItem
    key="/listing"
    onClick={[Function]}
  >
    LISTING
  </MenuItem>
</Menu>
`;
